src/main/java/org/openjdk/jextract/impl/FunctionalInterfaceBuilder.java

```diff
@@ -83,7 +83,15 @@ public interface %1$s {
     private void emitFunctionalFactory(String fiName) {
         appendIndentedLines("""
 
-            private static final MethodHandle UP$MH = %1$s.upcallHandle(%2$s.%3$s.class, "apply", $DESC);
+            static MethodHandle upcallHandle(Class<?> fi, String name, FunctionDescriptor fdesc) {
+                try {
+                    return MethodHandles.lookup().findVirtual(fi, name, fdesc.toMethodType());
+                } catch (ReflectiveOperationException ex) {
+                    throw new AssertionError(ex);
+                }
+            }
+
+            private static final MethodHandle UP$MH = upcallHandle(%2$s.%3$s.class, "apply", $DESC);
```

Comment:
Maybe you can inline the arguments into the implementation, since they seem to be constant?

```suggestion
            static MethodHandle upcallHandle() {
                try {
                    return MethodHandles.lookup().findVirtual(%2$s.%3$s.class, "apply", $DESC.toMethodType());
                } catch (ReflectiveOperationException ex) {
                    throw new AssertionError(ex);
                }
            }

            private static final MethodHandle UP$MH = upcallHandle();
```